kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -437,8 +425,8 @@ public void run() {
 
     @Override
     public long getDelay(TimeUnit unit) {
-      long delayMs = waitExpirationMs - config.getTime().milliseconds();
-      return unit.convert(delayMs, TimeUnit.MILLISECONDS);
+      Duration delay = Duration.between(clock.instant(), waitExpiration);
```

Comment:
i dunno, i'd probably wouldn't bother with the `Duration` etc stuff, i'd just work in ms internally. comment not intended to be actioned.

Reply:
That's an anti-pattern in Java (8+), due to the excellent standard library support for semantic date/times.